Modules/_abc.c

```diff
@@ -453,24 +515,23 @@ _abc__abc_init(PyObject *module, PyObject *self)
     if (PyType_Check(self)) {
         PyTypeObject *cls = (PyTypeObject *)self;
         PyObject *dict = _PyType_GetDict(cls);
-        PyObject *flags = PyDict_GetItemWithError(dict,
-                                                  &_Py_ID(__abc_tpflags__));
-        if (flags == NULL) {
-            if (PyErr_Occurred()) {
-                return NULL;
-            }
+        PyObject *flags = NULL;
+        int flags_result = PyDict_GetItemRef(dict, &_Py_ID(__abc_tpflags__), &flags);
```

Comment:
I think that has one behavioral difference with the existing code: if `__abc_tpflags__` is present but not an `int`, the old code will still delete it from the type's dict, while your suggested code won't. Do you think it's worth preserving that?

Reply:
Oh, I missed that. Yeah, I think that's worth preserving.

Reply:
In that case, I think it's simpler on balance to keep the overall structure mostly the same and add the missing `Py_DECREF()`. I'll update the PR soon and you can let me know what you think.

Another option is to delete from the dict before checking if it's an int, but that would cause a behavioral difference in the case where the user provides a bad int value for `__abc_tpflags__`.

Reply:
Or just replace `PyDict_GetItemRef` with `PyDict_Pop`.

There were two motivations for my suggestion above:

1) Having `Py_DECREF()` spread across the code makes it hard to tell if we're properly reference counting.
2) I find the `flags_result == 1` or `flags_result == 0` types more confusing than something like `flags == NULL` because the meaning of 0 vs. 1 depends on the specific function you are calling (as opposed to `-1`, which consistently means error).